library/Zend/Validate/NotEmpty.php

```diff
@@ -87,11 +87,11 @@ class Zend_Validate_NotEmpty extends Zend_Validate_Abstract
      */
     public function __construct($options = null)
     {
+        $func_args = func_get_args();
         if ($options instanceof Zend_Config) {
             $options = $options->toArray();
         } else if (!is_array($options)) {
-            trigger_error("PHP 7.2 Compatibility Alert:\n\tERROR: Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value.".sprintf(" (%s::%s)", __FILE__, __LINE__)."\n\t".implode("\n\t", array_map(function ($item) { return sprintf("%s::%s", $item['file'], $item['line']); }, debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS))), E_USER_WARNING);
-            $options = func_get_args();
+            $options = $func_args;
```

Comment:
@gfaza  Just curious - does the static code analysis warning go away when the call to `func_get_args()` is moved to the very beginning of the function?

Reply:
yup.  the full error message from the analysis is more like
```
FILE: /Users/frederic.zapata-alma/Projects/zf1-future/library/Zend/Validate/NotEmpty.php
FOUND 1 ERROR AFFECTING 1 LINE
 93 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter "$options" was changed on line\n91.
```
... but the messages often span multiple lines, and i ended up truncating to the end of the first sentence or end of the first line of the error.
This one goes away, b/c the original param is no longer threatened with being changed between the beginning of the call and our call to `func_get_args`.

Reply:
awesome 👌   love the approach, and glad the analyzer is smart enough to silence the error in that case.